network/manager.go

```diff
@@ -216,7 +216,14 @@ func (mgr *Manager) StartConnector() {
 // StartListener will start a listener on a given port.
 func (mgr *Manager) StartListener(stop <-chan struct{}) {
 	mgr.wg.Add(1)
-	go handleListening(mgr.log, mgr.wg, mgr.cfg, mgr, stop)
+	go handleListening(mgr.log, mgr.wg, mgr.cfg, mgr,
+		func(addr *net.TCPAddr) (Listener, error) {
+			listener, err := net.ListenTCP("tcp", addr)
+			if err != nil {
+				return nil, err
+			}
+			return &TcpListener{listener: listener}, nil
+		}, stop)
```

Comment:
This is really ugly, can you please put the function definition somewhere else?

Reply:
Done.